### PR TITLE
Fix Windows wheels

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:** We add tests for Windows Wheels, fix ill-defined caching, and set the proper backend for LKokkos wheels.

**Description of the Change:** Move tests to happen after patching the wheels, set a proper key for cache (compatible with Windows system), and set environment variables to set the proper backend to be built.
Building the LKokkos wheels is out-of-scope for this PR, and will be revisited when sensible.

**Benefits:** Proper wheels tested and working as intended.

**Possible Drawbacks:** For now we will not be building wheels for LKokkos on Windows.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by AmintorDusko** [comment]
[sc-59304]

File: .github/workflows/wheel_win_x86_64.yml [hunk 1]
```diff
@@ -108,11 +103,11 @@ jobs:
       - name: Checkout PennyLane-Lightning
         uses: actions/checkout@v4
 
-      - name: Copy cached libraries #Update when merging to pennylane-lightning
+      - name: Copy cached libraries
         if: steps.kokkos-cache.outputs.cache-hit == 'true'
         run: |
           Copy-Item -Path "D:\a\install_dir\${{ matrix.exec_model }}\" `
-                    -Destination "D:\a\Lightning-Unification\Lightning-Unification\Kokkos" -Recurse -Force
+                    -Destination "D:\a\pennylane-lightning\pennylane-lightning\Kokkos" -Recurse -Force
```
Comment (vincentmr) [inline .github/workflows/wheel_win_x86_64.yml:110]: Woh, that looks old!

File: .github/workflows/wheel_win_x86_64.yml [hunk 2]
```diff
@@ -160,6 +151,36 @@ jobs:
...
+
+      - name: Test wheels
+        run: |
+          python -m ensurepip --upgrade
+          python -m pip install setuptools
+          python -m pip install -r requirements-tests.txt
+          if ('${{ matrix.pl_backend }}' -eq 'lightning_kokkos'){
+            $SKIP_COMPILATION=true
+            $PL_BACKEND="lightning_qubit"
+            python -m pip install -e . -vv
```
Comment (vincentmr) [inline .github/workflows/wheel_win_x86_64.yml:173]: I think we need to either `export` the variables or set them on a single line
```suggestion
            SKIP_COMPILATION=true \
            PL_BACKEND="lightning_qubit" \
            python -m pip install -e . -vv
```

**Review by vincentmr — COMMENTED** [review]
We'll just need to update the CHANGELOG.

**Review by vincentmr — APPROVED** [review]
Looks good to me, thanks @AmintorDusko .

File: .github/workflows/wheel_win_x86_64.yml [hunk 3]
```diff
@@ -164,6 +155,36 @@ jobs:
...
+          }
+          cd wheelhouse
```
Comment (maliasadi) [inline .github/workflows/wheel_win_x86_64.yml:179]:
```suggestion
          pushd wheelhouse
```

File: .github/workflows/wheel_win_x86_64.yml [hunk 4]
```diff
@@ -164,6 +155,36 @@ jobs:
...
+          $wheels = Get-ChildItem "./" -Filter *.whl
+          foreach ($i in $wheels){
+            python -m pip install $i.Name
+          }
+          cd ..
```
Comment (maliasadi) [inline .github/workflows/wheel_win_x86_64.yml:184]:
```suggestion
          popd
```

File: .github/workflows/wheel_win_x86_64.yml [hunk 5]
```diff
@@ -207,4 +228,4 @@ jobs:
         with:
           user: __token__
           password: ${{ secrets.TEST_PYPI_API_TOKEN }}
-          repository-url: https://test.pypi.org/legacy/
+          repository-url: https://test.pypi.org/legacy/
```
Comment (maliasadi) [inline .github/workflows/wheel_win_x86_64.yml:231]:
```suggestion
          repository-url: https://test.pypi.org/legacy/

```

**Review by maliasadi — APPROVED** [review]
Thanks @AmintorDusko :raised_hands:

**Pull request merged**